### PR TITLE
Reservations RP 2018-06-01 api version

### PR DESCRIPTION
**Pull request by juhee0202**
Link to REST API spec PR: https://github.com/Azure/azure-rest-api-specs/pull/3149

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by juhee0202** [comment]
I am unsure how to complete the last two steps in SDK Generation Guidelines.
generate.cmd file does not contain any version info regarding AutoRest. what exactly should I update in generate.cmd file?
*.csproj and AssemblyInfo.cs does contain some versions (1.0.0.0). What value should I update this to? I am not sure how the initial version of 1.0.0.0 was created, but the new api version that we are creating for the .NET SDK is 2018-11-01. Is "assembly version" something different from "api version"?

**Comment by dsgouda** [comment]
1. Create a generate.ps1 similar to [this]() and use that instead of generate.cmd
2. You should update the csproj PackageVersion to `1.0.1-preview` and the `AssemblyFileVersion` in AssemblyInfo.cs to `1.0.1.0`

**Comment by juhee0202** [comment]
@dsgouda is there a reason why you specifically chose PackageVersion as 1.0.1-preview and AssemblyFileVersion as 1.0.1.0 ?
I am also working in CLI, and the CLI version is 0.2.0. Are they completely unrelated or should these two match?

**Comment by dsgouda** [comment]
CLI is unrelated. The current released version for sdk is `1.0.0-preview` if you check the csproj file. Since you are only making additive changes, bump the version before you publish a new package

**Comment by juhee0202** [comment]
I see. One more question, why not 1.0.1? Why do we have the preview suffix for the PackageVersion?

**Comment by dsgouda** [comment]
I am afraid that decision is not up to me. You will have to decide whether you are ready to go stable and mark these packages as stable by removing the tag.
If the REST spec API is in preview, the package version MUST also be in preview.

**Comment by juhee0202** [comment]
@dsgouda I was able to generate code using generate.ps1 as you suggested, but when I run this, it generates the code using previous API version 2017-11-01. Is this because the REST api has not been merged yet to the public repo?

**Comment by dsgouda** [comment]
@juhee0202 that's correct.

**Comment by juhee0202** [comment]
I have made the version changes (1.0.1-preview). however, the generated code in this repo has not been updated yet because the rest api specs have not yet been merged to public repo.

**Comment by juhee0202** [comment]
REST API specs have been merged to "preview".
I generated code using generate.ps1 script that I created. Updated the PR with
- AzSdk.RP.props 
- SdkInfo_AzureReservationAPI.cs

**Review by dsgouda — CHANGES_REQUESTED** [review]
A public type is being deleted in this PR. Please bump the assembly version to `1.1.0-preview`

**Comment by dsgouda** [comment]
@juhee0202 Please linked the merged REST spec PR

**Comment by dsgouda** [comment]
PR https://github.com/Azure/azure-rest-api-specs/pull/3149

**Comment by juhee0202** [comment]
Hi @dsgouda I was not able to generate code using the following command
Start-AutoRestCodeGeneration -ResourceProvider "reservations/resource-manager" -AutoRestVersion "latest"

Is there a step that I am missing to run "Start-AutoRestCodeGeneration"?

**Comment by juhee0202** [comment]
update: pulled from latest upstream branch (psSdkJson6) and ran the Generate.ps1 script and it worked.

**Review by dsgouda — APPROVED** [review]
LGTM will merge on CIs passing

**Comment by shahabhijeet** [comment]
@juhee0202 you need to get on-boarded to the GitHub org and that will kick off the CI job

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Can you update the package release notes for this release

**Comment by juhee0202** [comment]
https://cla.opensource.microsoft.com/ is a link to a PDF file that I can only read, but not sign. Do I need to sign the agreement somewhere for the license/cla to continue?

**Comment by juhee0202** [comment]
@shahabhijeet I have added changelog.md to SDKs/Reservations directory

File: src/SDKs/Reservations/Management.Reservations/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Reserved Instance Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Reserved Instance Resources.")]
 
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyVersion("1.1.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/Reservations/Management.Reservations/Properties/AssemblyInfo.cs]: @juhee0202 please do not update minor version on assemblyversion, unless you have a very specific reason to do that. We maintain semver rules, so you update AssemblyVersion only when you update the major version, else only AssemblyFileVersion is good enought

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Reservations/Management.Reservations/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (juhee0202) [reply src/SDKs/Reservations/Management.Reservations/Properties/AssemblyInfo.cs]: sure. I reverted back to 1.0.0.0

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**